pilot/pkg/networking/core/v1alpha3/networkfilter_test.go

```diff
@@ -147,6 +148,141 @@ func TestInboundNetworkFilterIdleTimeout(t *testing.T) {
 	}
 }
 
+func TestBuildOutboundNetworkFiltersWithSingleDestinationForTunnelingConfig(t *testing.T) {
+	type tunnelingConfig struct {
+		hostname string
+		usePost  bool
+	}
+
+	tunnelingEnabled := &networking.DestinationRule{
+		Host: "tunnel-proxy.com",
+		TrafficPolicy: &networking.TrafficPolicy{
+			Tunnel: &networking.TrafficPolicy_TunnelSettings{
+				Protocol:   "connect",
+				TargetHost: "example.com",
+				TargetPort: 8443,
+			},
+		},
+	}
+	tunnelingEnabledForSubset := &networking.DestinationRule{
+		Host: "tunnel-proxy.com",
+		Subsets: []*networking.Subset{
+			{
+				Name: "example-com-8443",
+				TrafficPolicy: &networking.TrafficPolicy{
+					Tunnel: &networking.TrafficPolicy_TunnelSettings{
+						Protocol:   "post",
+						TargetHost: "example.com",
+						TargetPort: 8443,
+					},
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                    string
+		destinationRule         *networking.DestinationRule
+		subset                  string
+		expectedTunnelingConfig *tunnelingConfig
+	}{
+		{
+			name:                    "tunneling_config should not be applied when destination rule and listener subsets do not match",
+			destinationRule:         tunnelingEnabledForSubset,
+			subset:                  "another-subset",
+			expectedTunnelingConfig: nil,
+		},
+		{
+			name:            "tunneling_config should be applied when hosts array contains exactly 1 host and destination rule has defined tunneling settings",
+			destinationRule: tunnelingEnabled,
+			expectedTunnelingConfig: &tunnelingConfig{
+				hostname: "example.com:8443",
+				usePost:  false,
+			},
+		},
+		{
+			name: "tunneling_config should be applied when hosts array contains exactly 1 host and " +
+				"destination rule has defined tunneling settings for a subset",
+			destinationRule: tunnelingEnabledForSubset,
+			subset:          "example-com-8443",
+			expectedTunnelingConfig: &tunnelingConfig{
+				hostname: "example.com:8443",
+				usePost:  true,
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			destinationRuleConfig := config.Config{
+				Meta: config.Meta{
+					GroupVersionKind: collections.IstioNetworkingV1Alpha3Destinationrules.Resource().GroupVersionKind(),
+					Name:             "tunnel-config",
+					Namespace:        "not-default",
+				},
+				Spec: tt.destinationRule,
+			}
+			cg := NewConfigGenTest(t, TestOptions{
+				ConfigPointers: []*config.Config{&destinationRuleConfig},
+				Services:       []*model.Service{buildServiceWithPort("example.com", 443, protocol.TLS, tnow)},
+			})
+			proxy := cg.SetupProxy(&model.Proxy{ConfigNamespace: "not-default"})
+
+			listeners := buildOutboundNetworkFiltersWithSingleDestination(cg.PushContext(), proxy, "", "",
```

Comment:
nit: `listeners` here is misleading; its not a listener but a Filter

Reply:
Fixed